### PR TITLE
#1-4Task. Popov Dmitriy

### PR DESCRIPTION
**Pull request by PopovDmitriy**

### CODE REVIEW

**Comment by PopovDmitriy** [comment]
> 
> https://travis-ci.com/github/PopovDmitriy/js-assignments

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -201,7 +202,20 @@ function extractEmails(str) {
  *
  */
 function getRectangleString(width, height) {
-    throw new Error('Not implemented');
+    let str = '';
+    for(var i = 1; i <= height; i++) {
```
Comment (FedorovD) [inline task/01-strings-tasks.js:206]: `var` is deprecated. It has no sense to use for browsers code even currently.

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -221,7 +223,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    return Number(value) ? Number(value) : def;
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:226]: not correct, if value = '0' ?

File: task/04-arrays-tasks.js [hunk 3]
```diff
@@ -102,7 +110,11 @@ function getArrayOfStrings(arr) {
  *    [ false, 0, NaN, '', undefined ]   => [ ]
  */
 function removeFalsyValues(arr) {
-   throw new Error('Not implemented');
+   let newArr = [];
+   arr.map(item => {
+      Boolean(item) ? newArr.push(item) : null
+   })
+   return newArr;
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:118]: use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: task/04-arrays-tasks.js [hunk 4]
```diff
@@ -249,7 +270,11 @@ function getMovingSum(arr) {
  * [ "a" ] => []
  */
 function getSecondItems(arr) {
-   throw new Error('Not implemented');
+   let newArr = [];
+   arr.map((item, index) => {
+      index % 2 == 1 ? newArr.push(item) : null
+   });
+   return newArr;
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:278]: can be done with `filter` method only

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -415,7 +461,15 @@ function toStringList(arr) {
  *      { country: 'Russia',  city: 'Saint Petersburg' }
  */
 function sortCitiesArray(arr) {
-   throw new Error('Not implemented');
+   return arr.sort((a, b) => {
+      if (a.country > b.country) {
+         return 1;
+      } else if (a.country == b.country){
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:467]: operator `===` is always preferable

**Comment by FedorovD** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Pull request closed without merging**